FinSearchUnified/Tests/Bundle/SearchBundleFindologic/FacetHandler/TextFacetHandlerTest.php

```diff
@@ -0,0 +1,360 @@
+     *
+     * @param array $filterData
+     * @param ValueListFacetResult $facetResult
+     * @param ConditionInterface|null $condition
+     */
+    public function testFacetModeForListFacetResult(
+        array $filterData,
+        ValueListFacetResult $facetResult,
+        ConditionInterface $condition = null
+    ) {
+        $facet = new ProductAttributeFacet(
+            'vendor',
+            ProductAttributeFacet::MODE_VALUE_LIST_RESULT,
+            'vendor',
+            'Manufacturer'
+        );
+        $criteria = new Criteria();
+        if ($condition !== null) {
+            $criteria->addCondition($condition);
+        }
+
+        $filter = $this->generateFilter($filterData);
+
+        $facetHandler = new TextFacetHandler();
+        $result = $facetHandler->generatePartialFacet($facet, $criteria, $filter);
+
+        $this->assertEquals($facetResult, $result);
+    }
+
+    /**
+     * @dataProvider radioListDataProvider
+     *
+     * @param array $filterData
+     * @param RadioFacetResult $facetResult
+     * @param ConditionInterface|null $condition
+     */
+    public function testFacetModeForRadioFacetResult(
+        array $filterData,
+        RadioFacetResult $facetResult,
+        ConditionInterface $condition = null
+    ) {
+        $facet = new ProductAttributeFacet(
+            'vendor',
+            ProductAttributeFacet::MODE_RADIO_LIST_RESULT,
+            'vendor',
+            'Manufacturer'
+        );
+        $criteria = new Criteria();
+        if ($condition !== null) {
+            $criteria->addCondition($condition);
+        }
+
+        $filter = $this->generateFilter($filterData);
+
+        $facetHandler = new TextFacetHandler();
+        $result = $facetHandler->generatePartialFacet($facet, $criteria, $filter);
+
+        $this->assertEquals($facetResult, $result);
+    }
+
+    public function valueListDataProvider()
+    {
+        return [
+            'Facet with condition and filter' => [
+                [
+                    'name' => 'vendor',
+                    'display' => 'Manufacturer',
+                    'select' => 'single',
+                    'type' => 'label',
+                    'items' => [
+                        [
+                            'name' => 'FINDOLOGIC',
+                            'frequency' => 42
+                        ]
+                    ]
+                ],
+                new ValueListFacetResult(
+                    'product_attribute_vendor',
+                    true,
+                    'Manufacturer',
+                    [
+                        new ValueListItem(
+                            'FINDOLOGIC',
+                            'FINDOLOGIC',
+                            true
+                        )
+                    ],
+                    'vendor'
+                ),
+                new ProductAttributeCondition('vendor', ConditionInterface::OPERATOR_EQ, 'FINDOLOGIC'),
```

Comment:
Affects all conditions in this data provider:
Convert `'FINDOLOGIC'` into an array since it also will be like this in production.